packages/loader/container-loader/src/container.ts

```diff
@@ -1995,10 +2003,15 @@ export class Container
 		return pkg as IFluidCodeDetails;
 	}
 
-	private get client(): IClient {
+	private static setupClient(
+		containerId: string,
+		options?: ILoaderOptions,
```

Comment:
As a best practice methods should only take the data they need. Here the only thing that is of interest is the client property of ILoaderOptions.
Unfortunately, that is not typed. I don't know why and seems like we could improve. That fix would be a breaking change (and might catch some bugs); so not for now. Given that it is untyped I wouldn't change the signature from what you have - just a note.